### PR TITLE
[OSE-171][OSE-172] Implement Get and List submissions

### PR DESCRIPTION
**Pull request by andresuribe87**
This PR implements the list and get endpoints according to SIP6. Please note that there was a big refactoring of the tests file in order to:
* Provide more clarity into what's being created before exercising the behavior under test. 
* Cleaning up after every test, so the DB is in a clean state.

### CODE REVIEW

File: doc/swagger.yaml [hunk 1]
```diff
@@ -1931,7 +1950,7 @@ paths:
         name: request
         required: true
         schema:
-          $ref: '#/definitions/pkg_server_router.StoreKeyRequest'
+          $ref: '#/definitions/github.com_tbd54566975_ssi-service_pkg_server_router.StoreKeyRequest'
```
Comment (andresuribe87) [inline doc/swagger.yaml:1953]: I'm not sure why this happens. Curious if it's something to do with my version of swaggo?

Comment (decentralgabe) [reply doc/swagger.yaml:1953]: I don't think it really matters as long as it gets rendered properly

File: pkg/server/router/presentation.go [hunk 2]
```diff
@@ -265,7 +265,7 @@ func (pr PresentationRouter) CreateSubmission(ctx context.Context, w http.Respon
 }
 
 type GetSubmissionResponse struct {
-	Submission exchange.PresentationSubmission `json:"submission"`
+	Submission presentation.Submission
```
Comment (decentralgabe) [inline pkg/server/router/presentation.go:268]: why remove the json?

File: pkg/server/server_presentation_test.go [hunk 3]
```diff
@@ -254,34 +424,12 @@ func createPresentationDefinition(t *testing.T, pRouter *router.PresentationRout
...
-	knownJWK := crypto.PrivateKeyJWK{
-		KTY: "EC",
-		CRV: "P-256",
-		X:   "6HEz8SLP7NgHPGp0bElryiD7u3_cO1EmX-ngsV_yLsI",
-		Y:   "QlIYaYyDLxLkybDan9LOSkfGvjzZsrdgAb_nQr_Li5M",
-		D:   "7m6c2Axy9OWi7-d9hFVhmMe22vQTfQDL_pG-3WFsjzc",
-	}
-
-	signer, err := crypto.NewJWTSignerFromJWK(knownJWK.KID, knownJWK)
+	signer, err := crypto.NewJWTSigner("", private)
```
Comment (decentralgabe) [inline pkg/server/server_presentation_test.go:431]: why `""` ?

File: pkg/service/presentation/service.go [hunk 4]
```diff
@@ -201,10 +203,31 @@ func (s Service) GetSubmission(request GetSubmissionRequest) (*GetSubmissionResp
...
-		return nil, util.LoggingNewErrorf("error getting presentation submission: %s", request.ID)
+		return nil, errors.Wrap(err, "fetching from storage")
 	}
-	if storedSubmission == nil {
-		return nil, util.LoggingNewErrorf("presentation submission with id<%s> could not be found", request.ID)
+	return &GetSubmissionResponse{
+		Submission: Submission{
+			Status:                 storedSubmission.Status.String(),
+			PresentationSubmission: &storedSubmission.Submission,
+		}}, nil
```
Comment (decentralgabe) [inline pkg/service/presentation/service.go:212]: same here

**Review by decentralgabe — COMMENTED** [review]
looks good - small comments

**Review by decentralgabe — APPROVED** [review]

**Pull request merged**